clio-client/src/main/scala/org/broadinstitute/clio/client/commands/ClioCommand.scala

```diff
@@ -21,6 +21,16 @@ import shapeless.CNil
   */
 sealed trait ClioCommand
 
+sealed trait GetSchemaCommand extends ClioCommand {
+  def index: TransferIndex
+}
+
+sealed trait AddCommand extends ClioCommand {
+  def index: TransferIndex
+  def key: TransferKey
```

Comment:
If we really want to lock this down, we could add another `type` param to `TransferIndex` for key type, and then this could be `def key: index.KeyType`.